addons/account_invoice_import_ubl/models/account_move.py

```diff
@@ -174,8 +179,48 @@ def _decode_ubl_2_1(self, tree):
 
         return invoice_form.save()
 
+    def _get_ubl_values(self):
+        def format_monetary(amount):
+            # Format the monetary values to avoid trailing decimals (e.g. 90.85000000000001).
+            return float_repr(amount, self.currency_id.decimal_places)
+
+        def get_line_values(line):
+            return {
+                'note': _('Discount (%s %)') % line.discount if line.discount else None,
+                'taxes': line.tax_ids.compute_all(
+                    line.price_unit,
+                    quantity=line.quantity,
+                    product=line.product_id,
+                    partner=self.partner_id)['taxes'],
+            }
```

Comment:
Why not passing a list to the xml? `'invoice_line_values': [{...} for line in self.invoice_line_ids]`